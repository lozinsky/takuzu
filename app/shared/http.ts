export function getErrorResponse(error: unknown): unknown {
  if (error instanceof Error && error.name === 'AssertionError') {
    return new Response(null, { status: 400, statusText: error.message });
  }

  return error;
}
