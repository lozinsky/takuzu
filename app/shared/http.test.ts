import { assert, expect, test } from 'vitest';

import { getErrorResponse } from './http';

class AssertionError extends Error {
  constructor(message?: string) {
    super(message);

    this.name = 'AssertionError';
  }
}

test('returns response', async () => {
  const candidate = getErrorResponse(new AssertionError('Message'));

  assert(candidate instanceof Response);

  expect({ message: await candidate.text(), status: candidate.status }).toMatchSnapshot();
});

test.each([new Error('Message'), 'Message', '', 0, null, undefined])('returns error', (value) => {
  expect(getErrorResponse(value)).toMatchSnapshot();
});
