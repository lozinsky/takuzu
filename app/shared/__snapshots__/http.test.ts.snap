// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`returns error 1`] = `[Error: Message]`;

exports[`returns error 2`] = `"Message"`;

exports[`returns error 3`] = `""`;

exports[`returns error 4`] = `0`;

exports[`returns error 5`] = `null`;

exports[`returns error 6`] = `undefined`;

exports[`returns response 1`] = `
{
  "message": "Message",
  "status": 400,
}
`;
